watertap/tools/parameter_sweep.py

```diff
@@ -430,9 +666,17 @@ def parameter_sweep(model, sweep_params, outputs, results_file=None, optimize_fu
         # Update the model values with a single combination from the parameter space
         _update_model_values(model, sweep_params, local_values[k, :])
 
+        # Forced reinitialization of the flowsheet if enabled
+        if reinitialize_before_sweep:
+            assert reinitialize_function is not None
+            reinitialize_function(model, **reinitialize_kwargs)
+
+        # Attempt to actually solve the flowsheet
         try:
-            # Simulate/optimize with this set of parameters
-            optimize_function(model, **optimize_kwargs)
+            # Simulate/optimize with this set of parameter
+            results = optimize_function(model, **optimize_kwargs)
+            pyo.assert_optimal_termination(results)
+            # _force_exception(k) # DELETE_ME!!!
```

Comment:
Why have `try/except` block here at all, if we *expect* the `optimize_function` to return the result object? We can use `pyo.check_optimal_termination` instead.

Reply:
Based on our discussion, it was decided to keep it as a try/except block.